src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -92,7 +99,7 @@ export interface DiscoverGridProps {
   /**
    * Determines if data is currently loaded
    */
-  isLoading: boolean;
+  loadingState: DataLoadingState;
```

Comment:
@weltenwort 

> Not a request to change anything here, but a thought on data structure design 🤔 since it can't be loading and loading_more at the same time, a single field which indicates either loading, loading_more or idle could represent this in a way that makes invalid state unrepresentable.

@davismcphee 

> I like this idea too. Not a blocker, but I think it would be a clearer API for the component.

I did some refactoring in https://github.com/elastic/kibana/pull/163784/commits/3752fdc51c931743d622f2de0808cc5eeaf75de3 Is it close to what you meant?

Reply:
Agreed, looks great!